File: Java/Mass.java
import java.util.Random;

public class Mass {
    private final int[] mass;

    public Mass(int sizeOfMass)
    {
        Random rand = new Random();
        mass = new int[sizeOfMass];
        for (int i = 0; i < mass.length; i++) {
            mass[i] = rand.nextInt(0,100);
        }

        mass[5] = -1;
    }

    synchronized int[] GetMinElement(int start, int end)
    {
        int minValue = Integer.MAX_VALUE;
        int minIndex = 0;
        for (int i = start; i <= end; i++) {
            if (mass[i] < minValue) {
                minValue = mass[i];
                minIndex = i;
            }
        }

        return new int[]{minValue, minIndex};
    }
}

File: C#/Mass.cs
using System;

namespace LB1
{
    public class Mass
    {
        private int[] data;
        public int Length => data.Length;
        public Mass(int size)
        {
            data = new int[size];
                Random random = new Random();
            for (int i = 0; i < size ; i++)
            {
                data[i] = random.Next(0, 100);
            }

            data[5] = -1;
        }

        public int[] GetMinElement(int start, int end)
        {
            int minVal = int.MaxValue;
            int minIndex = start;

            for (int i = start; i <= end; i++)
            {
                if (data[i] < minVal)
                {
                    minVal = data[i];
                    minIndex = i;
                }
            }

            return new int[] { minVal, minIndex };
        }
    }

}
